File: Java/0448-find-all-numbers-disappeared-in-an-array.java
import java.util.Set;

class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> res = new ArrayList<Integer>();
        Set<Integer> hash = new HashSet<Integer>();
        for(int i:nums)
            hash.add(i);
        for(int i = 1; i <= nums.length; i++){
            if(!hash.contains(i))
                res.add(i);
        }
        return res;
    }
}

File: Python/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        res = []
        hash = set()
        for i in nums:
            hash.add(i)
        for i in range(1, len(nums) + 1):
            if i not in hash:
                res.append(i)
        return res